### DISCUSSION TITLE
Starting own managed operator

### DISCUSSION THREAD

**Discussion by mixicz** [nats-io/nsc#677] [Q&A] [answered]
I want to provide managed operator for teams to be able to manage imports and exports on their own accounts. But I found only [usage of managed operators](https://docs.nats.io/using-nats/nats-tools/nsc/managed) in NATS documentation.

How can I start my own managed operator? Is it a feature of deprecated account server? Is it provided by nats resolver? Should I implement my own? Are there some commercial options?

**Comment by aricart** [comment, marked as answer]
The managed operator requires an https://github.com/nats-io/nats-account-server - note that this is not currently developed. Instead you should simply create a server that receives self-signed accounts, then your server assigns the account limits based on who they are (you have to manage your business logic), and signs the account with an operator key, and then pushes the properly signed account to the nats resolver. That is if you need an automatic process. Why this? the only reason for this is to allow someone to manage their account and users - you simply just on-board the account. Unless you are selling a NATS cloud service, this feature has no use for you.

**Reply by mixicz** [reply]
Well, I'm not selling NATS cloud service, I just want to provide NATS cloud service to internal teams which is basically the same except the billing part. And yes, I want to have automated process in the end (we are starting with manual signing of account changes, but I'm exploring options for future). Idea behind automation is that we expect fair amount of import and export changes on accounts which would needlessly create bottlenecks and manual signing overhead. And it would also allow us to create reliable audit log of export and import changes on accounts, which would make our security guys happy.

Regarding interface between the account server and nsc, I wasn't able to find any documentation, but from your description I expect it to be very simple and I guess I should just refer to nsc sources for details?

**Reply by aricart** [reply]
I added an example here:

https://github.com/aricart/simple_accontserver

**Reply by mixicz** [reply]
Thank you very much, this will be very helpful.
